### PR TITLE
feat: Add simple cli tool

### PR DESCRIPTION
**Pull request by lwaddicor**
This adds a simple cli tool which enables users to easily make requests
using their specified server address and protocol.

The response will be returned as pretty json for both human and machine
consumption.

### CODE REVIEW

File: cmd/cli/main.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	proto := flag.String("proto", "", "Protocol e.g. sqp, tf2e")
+	flag.Parse()
+
+	l := log.New(os.Stderr, "", 0)
+
+	if *address == "" {
+		l.Println("No address provided")
+		flag.PrintDefaults()
+		os.Exit(1)
+
```
Comment (stevenh) [inline cmd/cli/main.go:24]:
```suggestion
```

File: cmd/cli/main.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	c, err := svrquery.NewClient(*proto, *address)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer c.Close()
+
+	r, err := c.Query()
+	if err != nil {
+		log.Fatal(err)
```
Comment (stevenh) [inline cmd/cli/main.go:41]: This breaks the defer

File: cmd/cli/main.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	b, err := json.MarshalIndent(r, "", "\t")
+	if err != nil {
+		log.Fatal(err)
```
Comment (stevenh) [inline cmd/cli/main.go:46]: This breaks the defer

Comment (lwaddicor) [reply cmd/cli/main.go:46]: Done

File: cmd/cli/main.go (hunk 2 above)
Comment (lwaddicor) [reply cmd/cli/main.go]: Done

File: cmd/cli/main.go (hunk 1 above)
Comment (lwaddicor) [reply cmd/cli/main.go]: Done

File: cmd/cli/main.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+	}
+
+	c, err := svrquery.NewClient(*proto, *address)
+	if err != nil {
+		l.Fatal(err)
+	}
+	defer c.Close()
+
+	if err = query(c); err != nil {
+		l.Println(err)
```
Comment (stevenh) [inline cmd/cli/main.go:39]: This wont trigger a non zero exit code which is expected due to the failure.

Comment (lwaddicor) [reply cmd/cli/main.go]: Changed to remove the defer

File: cmd/cli/main.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+		os.Exit(1)
+	}
+
+	if *proto == "" {
+		l.Println("No protocol provided")
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	c, err := svrquery.NewClient(*proto, *address)
```
Comment (stevenh) [inline cmd/cli/main.go:32]: Silly question why not push the this into the query method then the defer will work as expected and is more idiomatic?

Comment (lwaddicor) [reply cmd/cli/main.go:32]: Yeah that does make more sense. Done

**Comment by lwaddicor** [comment]
retrigger

File: README.md [hunk 6]
```diff
@@ -43,6 +43,12 @@ func main() {
 }
 ```
 
+CLU
```
Comment (darkliquid) [inline README.md:46]: CLI?
```suggestion
CLI
```

**Pull request merged**